contracts/Joint.sol

```diff
@@ -182,54 +187,190 @@ contract Joint {
+                }
+
+                (uint256 aPnL, uint256 bPnL) =
+                    getPnL(currentA, currentB, _investedA, _investedB);
+                emit PnL(aPnL, bPnL);
+            }
+
+            distributeProfit();
         }
+    }
 
+    function adjustPosition() external onlyProviders {
         // No capital, nothing to do
-        if (balanceOfA() == 0 && balanceOfB() == 0) {
+        if (balanceOfA() == 0 || balanceOfB() == 0) {
             return;
         }
 
         if (reinvest) {
-            createLP();
+            (investedA, investedB, ) = createLP();
             depositLP();
+        }
+    }
+
+    event PnL(uint256 tokenA, uint256 tokenB);
+    event SellToBalance(address sellToken, uint256 sellAmount);
+
+    function calculateSellToBalance(
+        uint256 currentA,
+        uint256 currentB,
+        uint256 startingA,
+        uint256 startingB
+    ) internal returns (address _sellToken, uint256 _sellAmount) {
+        if (startingA == 0 || startingB == 0) return (address(0), 0);
+
+        (uint256 percentReturnA, uint256 percentReturnB) =
+            getPnL(currentA, currentB, startingA, startingB);
+
+        // If returns are equal, nothing to sell
+        if (percentReturnA == percentReturnB) return (address(0), 0);
+
+        emit PnL(percentReturnA, percentReturnB);
+
+        (uint256 _AForB, uint256 _BForA) = getSpotExchangeRates();
+
+        uint256 numerator;
+        uint256 denominator;
+        if (percentReturnA > percentReturnB) {
+            _sellToken = tokenA;
+            numerator = currentA.sub(startingA.mul(currentB).div(startingB));
+            denominator = 1e18 + startingA.mul(_BForA).div(startingB);
         } else {
-            distributeProfit();
+            _sellToken = tokenB;
+            numerator = currentB.sub(startingB.mul(currentA).div(startingA));
+            denominator = 1e18 + startingB.mul(_AForB).div(startingA);
         }
+        _sellAmount = numerator.mul(1e18).div(denominator);
+        emit SellToBalance(_sellToken, _sellAmount);
     }
 
-    function createLP() internal {
-        IUniswapV2Router02(router).addLiquidity(
-            tokenA,
-            tokenB,
-            balanceOfA(),
-            balanceOfB(),
-            0,
-            0,
-            address(this),
-            now
-        );
+    function getPnL(
+        uint256 currentA,
+        uint256 currentB,
+        uint256 startingA,
+        uint256 startingB
+    ) internal pure returns (uint256 _a, uint256 _b) {
+        _a = currentA.mul(1e4).div(startingA);
+        _b = currentB.mul(1e4).div(startingB);
+    }
+
+    function getSpotExchangeRates()
+        public
+        view
+        returns (uint256 _AForB, uint256 _BForA)
+    {
+        (uint256 reserve0, uint256 reserve1, ) = pair.getReserves();
+        uint256 _0For1 = reserve0.mul(1e18).div(reserve1);
+        uint256 _1For0 = reserve1.mul(1e18).div(reserve0);
```

Comment:
always 1e18 or it depends on token decimals?

Reply:
should be decimals, but i hardcoded because i was having some trouble. will fix ...